.github/workflows/build-wheel-macos-x86_64.yaml

```
@@ -0,0 +1,345 @@
+              -DMLIR_DIR=$GITHUB_WORKSPACE/llvm-build/lib/cmake/mlir \
+              -DMHLO_DIR=$GITHUB_WORKSPACE/mhlo-build/lib/cmake/mlir-hlo \
+              -DMHLO_BINARY_DIR=$GITHUB_WORKSPACE/mhlo-build/bin \
+              -DEnzyme_DIR=$GITHUB_WORKSPACE/enzyme-build \
+              -DENZYME_SRC_DIR=$GITHUB_WORKSPACE/mlir/Enzyme \
+              -DLLVM_ENABLE_ZLIB=OFF \
+              -DLLVM_ENABLE_ZSTD=OFF \
+              -DLLVM_ENABLE_LLD=OFF
+
+        cmake --build quantum-build --target check-dialects compiler_driver
+
+    - name: Build wheel
+      run: |
+        PYTHON=python${{ matrix.python_version }} \
+        LLVM_BUILD_DIR=$GITHUB_WORKSPACE/llvm-build \
+        MHLO_BUILD_DIR=$GITHUB_WORKSPACE/mhlo-build \
+        DIALECTS_BUILD_DIR=$GITHUB_WORKSPACE/quantum-build \
+        RT_BUILD_DIR=$GITHUB_WORKSPACE/runtime-build \
+        ENZYME_BUILD_DIR=$GITHUB_WORKSPACE/enzyme-build \
+        make wheel
+
+    - name: Upload Wheel Artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: catalyst-macos_x86_64-wheel-py-${{ matrix.python_version }}.zip
+        path: dist/
+        retention-days: 14
+
+  test-wheels:
+    needs: [constants, catalyst-macos-wheels-x86-64]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-12]
+        arch: [x86_64]
+        python_version: ${{ fromJson(needs.constants.outputs.python_versions) }}
+
+    # To check all wheels for supported python3 versions
+    name: Test Wheels (Python ${{ matrix.python_version }}) on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout Catalyst repo
+      uses: actions/checkout@v3
+
+    - name: Download Wheel Artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: catalyst-macos_x86_64-wheel-py-${{ matrix.python_version }}.zip
+        path: dist
+
+    - name: Set up Python ${{ matrix.python_version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python_version }}
+
+    - name: Install Python dependencies
+      run: |
+        python${{ matrix.python_version }} -m pip install pytest pytest-xdist
+        python${{ matrix.python_version }} -m pip install tensorflow-cpu  # for autograph tests
+
+    - name: Install Catalyst
+      run: |
+        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl
+
+    - name: Install PennyLane Plugins
+      run: |
+        python${{ matrix.python_version }} -m pip install PennyLane-Lightning-Kokkos
+        python${{ matrix.python_version }} -m pip install amazon-braket-pennylane-plugin
+
+    - name: Run Python Pytest Tests
+      run: |
+        python${{ matrix.python_version }} -m pytest $GITHUB_WORKSPACE/frontend/test/pytest -n auto
+        python${{ matrix.python_version }} -m pytest $GITHUB_WORKSPACE/frontend/test/pytest --runbraket=LOCAL -n auto
```

Do we not support Kokkos here?

---

I shall add this after fixing https://github.com/PennyLaneAI/pennylane-lightning/issues/552
```suggestion
        python${{ matrix.python_version }} -m pytest $GITHUB_WORKSPACE/frontend/test/pytest -n auto
        python${{ matrix.python_version }} -m pytest $GITHUB_WORKSPACE/frontend/test/pytest --backend="lightning.kokkos" -n auto
        python${{ matrix.python_version }} -m pytest $GITHUB_WORKSPACE/frontend/test/pytest --runbraket=LOCAL -n auto
```

---

Ah I see, could we add a comment / todo here? :)

---

here [2460cf1](https://github.com/PennyLaneAI/catalyst/pull/347/commits/2460cf1bbadd8e89ab016485d76f96471c13a366) :slightly_smiling_face: